### PR TITLE
Added support for signing prekeys via unpublished_prekey_signature().…

### PR DESCRIPTION
**Pull request by anunayk**
… Added new error code OLM_NO_UNPUBLISHED_PREKEY. unpublished_prekey() now returns null if there is no unpublished prekey.

### CODE REVIEW

File: src/error.c [hunk 1]
```diff
@@ -40,7 +40,10 @@ const char * _olm_error_to_string(enum OlmErrorCode error)
 {
     if (error < (sizeof(ERRORS)/sizeof(ERRORS[0]))) {
         return ERRORS[error];
-    } else {
+    } else if (error == 255) {
+        return "NO_UNPUBLISHED_PREKEY";
+    }
+    else {
```
Comment (Ashoat) [inline src/error.c]: Can we merge these to keep the style consistent?

**Review by Ashoat — APPROVED** [review]
We'll also need `get_prekey_signature`... should that be in this PR or a follow-up one?

**Review by Ashoat — APPROVED** [review]

File: src/olm.cpp [hunk 2]
```diff
@@ -588,34 +597,45 @@ size_t olm_create_outbound_session(
...
+    _olm_ed25519_public_key signing_key;
     _olm_curve25519_public_key pre_key;
     _olm_curve25519_public_key one_time_key;
+    uint8_t raw_pre_key_signature[ED25519_SIGNATURE_LENGTH];
 
     olm::decode_base64(id_key, id_key_length, identity_key.public_key);
+    olm::decode_base64(s_key, s_key_length, signing_key.public_key);
     olm::decode_base64(ot_key, ot_key_length, one_time_key.public_key);
     olm::decode_base64(p_key, p_key_length, pre_key.public_key);
+    olm::decode_base64(p_sign, p_sign_length, raw_pre_key_signature);
```
Comment (Ashoat) [inline src/olm.cpp:636]: It doesn't appear that we're doing any bounds checking on the prekey signature. If a malicious user sends a extra-long prekey signature, would it be possible that the bounds of `raw_pre_key_signature` will overflow, leading to a potential buffer overflow attack?

Comment (anunayk) [reply src/olm.cpp:636]: Good catch, I'll fix that.

**Pull request merged**